app/controllers/favorites_controller.rb

```diff
@@ -3,36 +3,22 @@ class FavoritesController < ApplicationController
   before_action :login_required
 
   def index
+    @page_title = 'Favorites'
+
     return if params[:view] == 'bucket'
-    unless current_user.favorites.present?
+    unless (@favorites = current_user.favorites).exists?
```

Comment:
This should in fact be `present?` I think. If it doesn't exist, we haven't loaded anything and don't care anyway; if it does exist, we're just going to need to load it again on the next line. No reason to do the extra query.

Reply:
...right, it would cache xD I apologize for my overenthusiam there